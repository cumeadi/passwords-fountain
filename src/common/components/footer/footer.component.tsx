import { h } from 'preact';
import { TypedComponent } from '@/common/typings/prop-types';
import { Wrapper } from './footer.styles';
import { Text } from '@/modules/localisation/components/text';

export const Footer: TypedComponent<Props> = () => {
    return (
        <Wrapper>
            <Text>footer.createdBy</Text>
        </Wrapper>
    );
};

interface Props {}

Footer.propTypes = {};
