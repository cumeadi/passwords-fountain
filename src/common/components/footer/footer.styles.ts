import { h } from 'preact';
import { styled, setPragma } from 'goober';
import { media, stylelint, theme, typography } from '@/common/theme';
setPragma(h);

export const Wrapper = styled('footer')`
    display: grid;
    grid-template-areas:
        'socialLinks'
        'copyright';

    align-items: center;
    justify-content: center;
    grid-column-gap: ${theme.spacing.m18};
    grid-row-gap: ${theme.spacing.xs6};
    padding: ${theme.spacing.s12} 0;
    color: ${theme.colors.darkBlue};
    text-align: center;

    ${media.gte(theme.breakpoints.s480)(stylelint.css`
        ${typography.text18}
        grid-template-areas: 'copyright socialLinks';
    `)}
`;

export const CopyrightTextWrapper = styled('div')`
    grid-area: copyright;
`;

export const SocialLink = styled('a')``;

export const SocialLinksWrapper = styled('div')`
    display: grid;
    grid-area: socialLinks;
    justify-content: center;
    grid-auto-flow: column;
    grid-auto-columns: max-content;
    grid-column-gap: ${theme.spacing.m18};
`;
