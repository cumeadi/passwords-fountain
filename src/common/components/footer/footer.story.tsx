import { h, VNode } from 'preact';
import { Footer } from './footer.component';
import { withA11y } from '@storybook/addon-a11y';

export default {
    title: 'Footer',
    decorators: [withA11y],
};

export const defaultView = (): VNode => <Footer />;
