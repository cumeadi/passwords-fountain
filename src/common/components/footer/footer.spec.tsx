import '@testing-library/jest-dom/extend-expect';
import { h } from 'preact';
import { render } from '@testing-library/preact';
import { Footer } from './footer.component';

describe('Footer', () => {
    it('should render correctly', () => {
        const { asFragment } = render(<Footer />);

        expect(asFragment()).toMatchSnapshot();
    });
});
