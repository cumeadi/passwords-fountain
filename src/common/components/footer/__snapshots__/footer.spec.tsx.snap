// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
<DocumentFragment>
  <footer
    class="go2119715341"
  >
    <div
      class="go2046521801"
    >
      Created by Marcin Ko≈Çodziejczak
    </div>
    <div
      class="go2921248024"
    >
      <a
        class="go115679"
        href="https://github.com/kolodziejczakM"
      >
        <img
          alt="github icon"
          class="go115679"
          height="30"
          src="static-asset-stub"
          width="30"
        />
      </a>
      <a
        class="go115679"
        href="https://twitter.com/kolodziejczakMn"
      >
        <img
          alt="twitter icon"
          class="go115679"
          height="30"
          src="static-asset-stub"
          width="30"
        />
      </a>
    </div>
  </footer>
</DocumentFragment>
`;
